### PR TITLE
[DUOS-2243][risk=no] Add cache for external status checks

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Along with https://github.com/broadinstitute/terra-helmfile/pull/3714, this PR partially addresses https://broadworkbench.atlassian.net/browse/DUOS-2243

This PR adds caching to most external status checks to reduce the latency of consent's status check.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java [hunk 1]
```diff
@@ -19,6 +19,8 @@ public class ServicesConfiguration {
 
   private Integer poolSize = 10;
 
+  private Integer cacheExpireMinutes = 1;
```
Comment (nmalfroy) [inline src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java:37]: Is it weird to have this in minutes but you've got `timeoutSeconds` right above?

**Review by nmalfroy — APPROVED** [review]
This looks good.  Looking at the handler in `HttpClientUtil.getHttpResponse`, it seems like it's correctly consuming the entity response, so that's likely not where your memory leak is coming from

File: src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java:37]: I can certainly change it for consistency, but they are conceptually very different things.

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java:37]: Maybe some additional commentary around each value would be helpful

Comment (nmalfroy) [reply src/main/java/org/broadinstitute/consent/http/configurations/ServicesConfiguration.java:37]: Yeah, that's totally fair.  And there's likely no need to have a more exact cache duration than minutes.

Re docs, that's maybe more general than this PR.  But yeah, maybe java docs around the service config might be useful. Totally not a blocker to merge BTW

**Review by connorlbark — APPROVED** [review]
👍 works great locally for me, much snappier after the first call. code looks good, thorough comments and testing.

**Pull request merged**